llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16042,7 +16027,15 @@ SITargetLowering::shouldExpandAtomicRMWInIR(AtomicRMWInst *RMW) const {
     return AtomicExpansionKind::CmpXChg;
   }
   case AtomicRMWInst::FMin:
-  case AtomicRMWInst::FMax:
+  case AtomicRMWInst::FMax: {
+    Type *Ty = RMW->getType();
+
+    // LDS float and double fmin/fmax were always supported.
+    if (AS == AMDGPUAS::LOCAL_ADDRESS && (Ty->isFloatTy() || Ty->isDoubleTy()))
+      return AtomicExpansionKind::None;
```

Comment:
Can it be unsafe? I.e. with denorm handling?

Also 4 cases below now all integer but checking isFloatTy().

Reply:
No, LDS atomics are always well behaved. I think denormals always worked correctly even for the global cases too for fmin/fmax

Reply:
OK. What's with isFloatTy() below?

Reply:
Should be dead code, but in principle the unsafe handling would be handling the no-fine-grained-remote-memory case, not denormal mishandling

Reply:
> would be handling the no-fine-grained-remote-memory case, not denormal mishandling

Which don't matter for LDS

Reply:
The isFloatTy is not dead code, but it triggers a behavior change in umin/umax/smin/smax, so I'll fix this separately

Reply:
> The isFloatTy is not dead code, but it triggers a behavior change in umin/umax/smin/smax, so I'll fix this separately

Nevermind, this was only due to interpreting unsafeFP attribute for the integer case